pkg/registry/core/service/storage/rest.go

```diff
@@ -1158,6 +1176,44 @@ func allocateHealthCheckNodePort(service *api.Service, nodePortOp *portallocator
 	return nil
 }
 
+//FIXME: rename and merge with initNodePorts
+func (al *RESTAllocStuff) allocServiceNodePortsNew(service *api.Service, dryRun bool) (transaction, error) {
+	// The allocator tracks dry-run-ness internally.
```

Comment:
I think we can exit fast and avoid the nested if
```
if service.Spec.Type != api.ServiceTypeNodePort && service.Spec.Type != api.ServiceTypeLoadBalancer {
 return nil, nil
}
}

Reply:
If we allow nil `transaction` then we have to check for nil everywhere.  We could add a `nullTransaction` but I don't think it will be used more than this spot...